### DISCUSSION TITLE
How do buffs / debuffs work? How to make a consumable item (e.g. potion) give a chance on use for a stack-able debuff?

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#862] [Q&A] [answered]
I am trying to figure out what a 'risky potion' could look like where it has a chance for a bad effect / debuff to happen to the player where it prevents slamming back ~15 healing potions to face tank anything.

Can a consumable item / potion have an arbitrary % chance to trigger an effect (like a debuff). E.g. Heals X but has a 50% chance of -10% all player stats for 2 minutes. The effect (debuff) can stack / each debuff being applied consecutively / can have a whole bunch of the same but stacking debuff fill up the player debuff 'bar' if they consume / slam back ~15 potions to even potentially kill themselves with -10% All Stats (stacked up to) x 10.

Questions:
1. How can debuffs / buffs stack on top of each other?
2. Likewise can a buff / debuff be non-stacking?
3. Can a buff / debuff be non-stacking but extend the effect time limit; like 're-applying' a ~passive buff's time limit when it is running low but not double the effect?
4. How to apply -10% or +10% all player statistics / attributes in the Flare engine?
5. What are examples of what to do / what are the known solutions for buffs / debuffs?
6. What should not be done / what are known issues with buffs / debuffs?
7. What is the scope of ability / ~'developer options' available within the Flare engine for buffs / debuffs? What kinds / types of buffs / debuffs can be made in the Flare engine? Are there strict rules? Or a bunch of different types of options? Are there a bunch of different way to apply buffs / debuffs? What can be done in the Flare engine and what cannot be done currently in the Flare engine for buffs / debuffs? Is there a developer guide / manual for how to work with Flare engine buffs / debuffs?

**Comment by Danimal696** [comment]
From experience i can tell you debuff stacks, like poison or bleed; in powers/effects.txt:
_[effect]
id=heal
type=heal
icon=11
animation=animations/powers/heal.txt
can_stack=false

I can´t answer the others, except that most debufs are limited to giving damage.

**Comment by WithinAmnesia** [comment, marked as answer]
https://github.com/flareteam/flare-game/discussions/864 This may become this solution to this thread's intention of finding an improved system for supporting potentially hundreds of different types of potions / consumables without players avoiding cooldown within the Flare Engine; potentially ruining combat by becoming nigh-invincible.
